contrib/dockerd-rootless-setuptool.sh

```diff
@@ -0,0 +1,450 @@
+		systemctl --user start ${SYSTEMD_UNIT}
+		sleep 3
+		set +x
+	fi
+	set -x
+	systemctl --user --no-pager --full status ${SYSTEMD_UNIT}
+	DOCKER_HOST="unix://$XDG_RUNTIME_DIR/docker.sock" $BIN/docker version
+	systemctl --user enable ${SYSTEMD_UNIT}
+	set +x
+	INFO "Installed ${SYSTEMD_UNIT} successfully."
+	INFO "To control ${SYSTEMD_UNIT}, run: \`systemctl --user (start|stop|restart) ${SYSTEMD_UNIT}\`"
+	INFO "To run ${SYSTEMD_UNIT} on system startup, run: \`sudo loginctl enable-linger $(id -un)\`"
+	echo
+}
+
+# install (non-systemd)
+install_nonsystemd() {
+	INFO "systemd not detected, ${DOCKERD_ROOTLESS_SH} needs to be started manually:"
+	echo
+	echo "PATH=$BIN:/sbin:/usr/sbin:\$PATH ${DOCKERD_ROOTLESS_SH} ${DOCKERD_ROOTLESS_SH_FLAGS}"
+	echo
+}
+
+# CLI subcommand: "install"
+cmd_entrypoint_install() {
+	# requirements are already checked in init()
+	if [ -z "$SYSTEMD" ]; then
+		install_nonsystemd
+	else
+		install_systemd
+	fi
+
+	INFO "Make sure the following environment variables are set (or add them to ~/.bashrc):"
+	echo
+	if [ -n "$XDG_RUNTIME_DIR_CREATED" ]; then
+		echo "export XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR}"
+	fi
+	echo "export PATH=${BIN}:\$PATH"
+	echo "export DOCKER_HOST=unix://${XDG_RUNTIME_DIR}/docker.sock"
+	echo
+
+}
+
+# CLI subcommand: "uninstall"
+cmd_entrypoint_uninstall() {
+	# requirements are already checked in init()
+	if [ -z "$SYSTEMD" ]; then
+		INFO "systemd not detected, ${DOCKERD_ROOTLESS_SH} needs to be stopped manually:"
+	else
+		unit_file="${CFG_DIR}/systemd/user/${SYSTEMD_UNIT}"
+		set -x +e
+		systemctl --user stop ${SYSTEMD_UNIT}
+		systemctl --user disable ${SYSTEMD_UNIT}
+		rm -f ${unit_file}
+		set +x -e
+		INFO "Uninstalled ${SYSTEMD_UNIT}"
+	fi
+
+	INFO "This uninstallation tool does NOT remove Docker binaries and data."
+	INFO "To remove data, run: \`$BIN/rootlesskit rm -rf $HOME/.local/share/docker\`"
+}
+
+# text for --help
+usage() {
+	echo "Usage: ${ARG0} [OPTIONS] COMMAND"
+	echo
+	echo "A setup tool for Rootless Docker (${DOCKERD_ROOTLESS_SH})."
+	echo
+	echo "Documentation: https://docs.docker.com/engine/security/rootless/"
+	echo
+	echo "Options:"
+	echo "  -f, --force                Ignore rootful Docker (/var/run/docker.sock)"
+	echo "      --skip-iptables        Ignore missing iptables"
+	echo
+	echo "Commands:"
+	echo "  check        Check prerequisites"
+	echo "  install      Install systemd unit (if systemd is available) and show how to manage the service"
+	echo "  uninstall    Uninstall systemd unit"
+}
+
+# parse CLI args
+args=$(getopt -o hf --long help,force,skip-iptables -n $ARG0 -- "$@")
+getopt_status=$?
+if [ $getopt_status != 0 ]; then
+	usage
+	exit $getopt_status
+fi
+eval set -- "$args"
+while true; do
+	case "$1" in
+		-h | --help)
+			usage
+			exit 0
+			shift
+			;;
+		-f | --force)
+			FORCE_ROOTLESS_INSTALL=1
+			shift
+			;;
+		--skip-iptables)
+			SKIP_IPTABLES=1
+			shift
+			;;
+		--)
+			shift
+			break
+			;;
+		*)
+			break
+			;;
+	esac
+done
```

Comment:
I think this ends up being slightly more concise if we save `$1`, then `shift`, then `case`:

```suggestion
while [ "$#" -gt 0 ]; do
	arg="$1"
	shift
	case "$arg" in
		-h | --help)
			usage
			exit 0
			;;
		-f | --force)
			FORCE_ROOTLESS_INSTALL=1
			;;
		--skip-iptables)
			SKIP_IPTABLES=1
			;;
		--)
			break
			;;
		*)
			# XXX this means we missed something in our "getopt" arguments above!
			ERROR "Scripting error, unknown argument '$arg' when parsing script arguments."
			exit 1
			;;
	esac
done
```

Reply:
applied